### PR TITLE
Refactor and Simplify Documentation

### PR DESCRIPTION
**Pull request by danlooo**
This PR aims to make the documentation more readable and consistent. Everything from the old documentation is migrated to here if they are not redundant.

- One page per task, e.g. read or write. Group all file formats into one page
- Add explanation about the data types, e.g. Array vs Cube
- Add page on how to convert between other types: e.g. Raster
- Add example on how to read ArchGDAL
- Add example on mapCube that modifies input and output dimensions (i.e. lat and lon to region)
- Consistent naming of arrays and datasets
- Simplify get started: Only 2 lines of code to the first YAXArray

Please feel free to comment on further improvements to make this package as easy to use as possible.

### CODE REVIEW

**Comment by lazarusA** [comment]
the `etc` folder needs a better name.  Remember, the file structure will gives the actual final path to the resource. Hence, descriptive names are preferred.

**Comment by danlooo** [comment]
That's right. Its my working dir for stuff that is not integrated yet. I'll yet you know once this draft PR is ready

File: docs/src/index.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ hero:
     src: /logo.png
     alt: VitePress
   actions:
-    - theme: brand
-      text: Getting Started
-      link: /getting_started
+    - theme: alt
+      text: Get Started
```
[the remark this answers is not in the document]
Comment (lazarusA) [reply docs/src/index.md:15]: ahh.. is funny that Germany is the one in red, my preference. Is ok then... let's keep your suggestion 😄 .

**Comment by danlooo** [comment]
I'm done for my side with migrating the entire documentation to a consistent structure. Fell free to comment and merge @meggart @lazarusA  @felixcremer

**Review by lazarusA — APPROVED** [review]
Great job.

**Pull request merged**